### PR TITLE
Support Opening History Downloads + Empty History State

### PR DESCRIPTION
**Pull request by nlogozzo**
As per #453:
> so we can quickly search and open previously downloaded videos.

This PR makes it where the history will also save the path of a downloaded media. If that path still exists when the HistoryDialog is opened, an open button will be added to the row to allow the user to open the folder of that downloaded media:

image

### CODE REVIEW

File: NickvisionTubeConverter.GNOME/Controls/HistoryDialog.cs [hunk 1]
```diff
@@ -59,17 +70,31 @@ private HistoryDialog(Gtk.Builder builder, Gtk.Window parent, string iconName, D
...
                 DownloadAgainRequested?.Invoke(this, pair.Key);
             };
-            row.AddSuffix(button);
-            row.SetActivatableWidget(button);
+            row.AddSuffix(downloadButton);
+            if(File.Exists(pair.Value.Path))
+            {
+                var openButton = Gtk.Button.New();
+                openButton.SetIconName("document-open-symbolic");
+                openButton.SetTooltipText(_("Open"));
```
Comment (fsobolev) [inline NickvisionTubeConverter.GNOME/Controls/HistoryDialog.cs:87]: Does it open file or folder with file selected? If the former, then "play" icon would be better. If the latter, I would change tooltip to "Open in Folder".

**Review by fsobolev — CHANGES_REQUESTED** [review]

**Comment by fsobolev** [comment]
Also I think buttons should be in the opposite order: open first, then download again, so download again buttons will be in one column.

**Comment by nlogozzo** [comment]
Also added an empty history state:
image

File: NickvisionTubeConverter.GNOME/Controls/HistoryDialog.cs (hunk 1 above)
Comment (nlogozzo) [reply NickvisionTubeConverter.GNOME/Controls/HistoryDialog.cs:83]: it's the former so I will change to play icon

**Review by fsobolev — APPROVED** [review]

**Pull request merged**